src/formElementPlugins/listBuilder/src/ApplicationView.tsx

```diff
@@ -0,0 +1,339 @@
+import React, { useEffect, useState } from 'react'
+import {
+  Button,
+  Modal,
+  Form,
+  Segment,
+  Table,
+  TableHeaderCell,
+  TableCell,
+  Card,
+  Icon,
+  Label,
+} from 'semantic-ui-react'
+import { ApplicationViewProps } from '../../types'
+import { ResponseFull } from '../../../utils/types'
+import { TemplateElement, TemplateElementCategory } from '../../../utils/generated/graphql'
+import ApplicationViewWrapper from '../../ApplicationViewWrapper'
+import strings from '../constants'
+
+enum DisplayType {
+  CARDS = 'cards',
+  TABLE = 'table',
+}
+
+interface InputResponseField {
+  isValid?: boolean
+  value: ResponseFull
+}
+
+type ListItem = { [code: string]: InputResponseField }
+
+const ApplicationView: React.FC<ApplicationViewProps> = ({
+  element,
+  parameters,
+  onSave,
+  onUpdate,
+  Markdown,
+  initialValue,
+  currentResponse,
+  applicationData,
+  allResponses,
+}) => {
+  const { isEditable } = element
+  const {
+    label,
+    description,
+    createModalButtonText = strings.BUTTON_LAUNCH_MODAL,
+    modalText,
+    addButtonText = strings.BUTTON_ADD,
+    updateButtonText = strings.BUTTON_UPDATE,
+    deleteItemText = strings.BUTTON_DELETE,
+    inputFields,
+    displayFormat = getDefaultDisplayFormat(inputFields),
+    displayType = DisplayType.CARDS,
+  } = parameters
+
+  const [currentInputResponses, setCurrentInputResponses] = useState<ListItem>(
+    resetCurrentResponses(inputFields)
```

Comment:
anther smallTidy up (being a-little bit picky), resetCurrentResponse, setOpen(false) and selectedListItem(null) seem to be coupled ? up to you if you want either combine their state or make a helper that resets them all at once, or neither (it should be ok as it is)

Reply:
Sure, I made a `resetModalState` method:
f31b871a
382b15af
(Sorry, two commits)